client/my-sites/comments/comment/index.jsx

```diff
@@ -38,13 +39,38 @@ export class Comment extends Component {
 	};
 
 	state = {
+		hasScrolled: false,
 		isEditMode: false,
 		isReplyVisible: false,
 	};
 
 	componentWillReceiveProps( nextProps ) {
 		const { isBulkMode: wasBulkMode } = this.props;
-		const { isBulkMode } = nextProps;
+		const { commentId, isBulkMode, isLoading } = nextProps;
+		const { hasScrolled } = this.state;
+
+		if (
+			! hasScrolled &&
+			! isLoading &&
+			!! window &&
+			`#comment-${ commentId }` === window.location.hash
+		) {
+			const commentNode = ReactDom.findDOMNode( this.commentCard );
+			const commentOffsetTop = commentNode.offsetTop - 56;
+			window.requestAnimationFrame( () =>
```

Comment:
Should we give the page a chance to load all comments before scrolling down?

Reply:
Looks like we're queueing multiple `scrollTo` actions. I'd recommend changing the `hasScrolled` state into `isScrolling`. In order for this to work though we'll want to delay this call until comments have loaded.

eg:

```
if (
 			! isScrolling &&
 			! isLoading &&
 			!! window &&
 			`#comment-${ commentId }` === window.location.hash
		) {
     this.setState( { 'isScrolling', true } );
     //... queue scrollTo
}
```

Reply:
Also note that scrollTo uses RAF internally so I think we can avoid that here.